.github/workflows/onPRClose.yml

```diff
@@ -3,41 +3,75 @@ on:
   pull_request:
     types: [closed]
 jobs:
-  extractIssueNumber:
-    name: Extract Issue Number
+
+  get-pr-number:
+    name: Get Current PR Number
+    runs-on: ubuntu-latest
+    outputs: 
+      pr-number: ${{ steps.condval.outputs.value }}
+    steps:
+      - uses: 3drepo/action-cond@v1
+        id: condval
+        with:
+          cond: ${{ github.event_name == 'pull_request' }}
+          if_true: ${{ github.event.number }}
+          if_false: ${{ github.event.pull_request.number }}
+
+  get-issue-number:
+    name: Get Current Issue Number
+    needs: [ get-pr-number ]
     runs-on: ubuntu-latest
     outputs: 
       issue-number: ${{ steps.commitMsgParser.outputs.issue-number }}
+      content-id: ${{ steps.commitMsgParser.outputs.content-id }}
     steps:
       - name: Extract Issue Number from PR
-        uses: carmenfan/extract-issue-number-from-PR-merge@v1.1
+        uses: 3drepo/extract-issue-number-from-PR@v1
         with:
-          pr: ${{ github.event.number }}
+          pr: ${{ needs.get-pr-number.outputs.pr-number }}
           base: staging
           github-token: ${{ secrets.GITHUB_TOKEN }}
         id: commitMsgParser
-  tagOnPRMerge:
+
+  tag-on-pr-merge:
     name: Tag merged issues as In staging if applicable
     runs-on: ubuntu-latest
-    needs: extractIssueNumber
+    needs: get-issue-number
     steps:
       - name: Apply in staging label
-        if: needs.extractIssueNumber.outputs.issue-number
+        if: needs.get-issue-number.outputs.issue-number
         uses: carmenfan/any-issue-labeller@v1.1
         with:
-          issue-number: ${{ needs.extractIssueNumber.outputs.issue-number }}
+          issue-number: ${{ needs.get-issue-number.outputs.issue-number }}
           label: 'in staging'
           github-token: ${{ secrets.GITHUB_TOKEN }}
-  destroy:
-    name: Call Azure Pipeline
+
+  update-project-status:
+    name: Update status of the issue in the project
+    runs-on: ubuntu-latest
+    needs: get-issue-number
```

Comment:
Similar to above, Unless i've mistaken, I could've closed the issue without merging and this would erroneously tagged this as done.

Reply:
But this case is slightly different, we want to set it to done if it's been merged, we don't care about whether it's been merged to staging.